### PR TITLE
Correct suffix for solution tracers

### PR DESCRIPTION
**Pull request by svenn-t**
Needed in (draft) PR: https://github.com/OPM/opm-simulators/pull/5268.

### CODE REVIEW

File: opm/output/eclipse/Summary.cpp [hunk 1]
```diff
@@ -3865,7 +3937,15 @@ namespace Evaluator {
...
         }
 
-        const auto tracer_name = normKw.substr(4);
+        int istart;
+        if (normKw[4] == 'F' || normKw[4] == 'S') {
+            istart = 5;
+        }
+        else {
+            istart = 4;
+        }
```
[the remark this answers is not in the document]
Comment (svenn-t) [reply opm/output/eclipse/Summary.cpp]: I like the simplification so I changed the code. Thanks!

**Comment by svenn-t** [comment]
> There is one more instance of the [magic number 4](https://github.com/OPM/opm-common/blob/5ec05f463223ce90ad8bc864b25b948bf5cbf32e/opm/output/eclipse/Summary.cpp#L879) in related to tracers in `Summary.cpp`. This constant may or may not be important for this work–I don't know off hand–but we should at least add a comment for why we don't need to adjust it if it is not.
> 
> Other than that, this looks good to me.

The magic number 4 in `ratetracer()` actually works because for free and solution tracers the summary vectors are stored with a `"F"` or `"S"` prefix in the `tracer_name`. So for example the summary vector `WTPRFX11` is stored with `tracer_name = FX11` to get the free part of a tracer with actual name `X11`.

**Comment by akva2** [comment]
in that case something like
```
const auto prefix_len = std::string_view{"WTPR}.size();
```
is a lot more expressive than the magic number.

**Comment by svenn-t** [comment]
> in that case something like
> 
> ```c++
> constexpr auto prefix_len = std::strlen("WTPR");
> ```
> 
> is a lot more expressive than the magic number.

I agree, magic numbers are not the ideal. The `ratetracer()` function is used for all well-related tracer keywords (`WTPC`, `FTPT`, etc), so checking the length of a specific keyword doesn't work here, at least. But maybe it can be defined somewhere else, in the way you suggest, and put into the `args` input? I'm not so familiar with how everything works in the `Summary.cpp` file.

**Comment by akva2** [comment]
right, if it applies to more keywords something ala
````c++
constexpr auto prefix_len = 4; // 'WTPRxxx, FTPFxxx, etc
```
conveys the point just as well.

**Comment by svenn-t** [comment]
Agreed, I have updated the code. Thanks!

**Comment by totto82** [comment]
jenkins build this  opm-simulators=5268 please

**Comment by totto82** [comment]
jenkins build this opm-simulators=5268 please

**Comment by totto82** [comment]
jenkins build this opm-simulators=5268 please

**Comment by bska** [comment]
benchmark please opm-simulators=5268

**Comment by ytelses** [comment]
Benchmark result overview: 




| Test    | Configuration                                     | Relative |
| ------- | ------------------------------------------------- | -------- |
| opm-git | OPM Benchmark: drogon - Threads: 1                | 1.007    |
| opm-git | OPM Benchmark: drogon - Threads: 8                | 0.995    |
| opm-git | OPM Benchmark: punqs3 - Threads: 1                | 1.013    |
| opm-git | OPM Benchmark: punqs3 - Threads: 8                | 0.989    |
| opm-git | OPM Benchmark: smeaheia - Threads: 1              | 1.014    |
| opm-git | OPM Benchmark: smeaheia - Threads: 8              | 1.006    |
| opm-git | OPM Benchmark: spe10_model_1 - Threads: 1         | 1.004    |
| opm-git | OPM Benchmark: spe10_model_1 - Threads: 8         | 1        |
| opm-git | OPM Benchmark: flow_mpi_extra - Threads: 1        | 0.997    |
| opm-git | OPM Benchmark: flow_mpi_extra - Threads: 8        | 1.001    |
| opm-git | OPM Benchmark: flow_mpi_norne - Threads: 1        | 0.987    |
| opm-git | OPM Benchmark: flow_mpi_norne - Threads: 8        | 1.001    |
| opm-git | OPM Benchmark: flow_mpi_norne_4c_msw - Threads: 1 | 1.013    |
| opm-git | OPM Benchmark: flow_mpi_norne_4c_msw - Threads: 8 | 0.993    |
* Speed-up = Total time master / Total time pull request. Above 1.0 is an improvement. *



View result details @ https://www.ytelses.com/opm/?page=result&id=2505

**Comment by svenn-t** [comment]
> Is this PR itself independent of the downstream PR [OPM/opm-simulators#5268](https://github.com/OPM/opm-simulators/pull/5268), or must the two be reviewed and merged in concert?

I have not tried to compile changes in this PR with the master version of opm-simulators, so I don't know if any of the tests will fail or other unexpected errors will occur. Should mostly be extensions of summary file output in this PR so it will perhaps be fine to treat this PR independently.

**Comment by bska** [comment]
> > Is this PR itself independent of the downstream PR [OPM/opm-simulators#5268](https://github.com/OPM/opm-simulators/pull/5268), or must the two be reviewed and merged in concert?
> 
> I have not tried to compile changes in this PR with the master version of opm-simulators, so I don't know if any of the tests will fail or other unexpected errors will occur. Should mostly be extensions of summary file output in this PR so it will perhaps be fine to treat this PR independently.

Understood.  In that case I'll try a separate build check just for this PR and review/merge if that check goes through.

**Comment by bska** [comment]
jenkins build this please

**Review by bska — APPROVED** [review]
Thanks a lot.  This looks good to me now and I'll merge into master.

**Pull request merged**